daemon/gpu/gpu.go

```diff
@@ -0,0 +1,40 @@
+package gpu // import "github.com/docker/docker/daemon/gpu"
+
+import (
+	"fmt"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+var (
+	vendors map[string]SpecHandler
+)
+
+// SpecHandler modifies an OCI runtime spec to enable access to GPU devices.
+type SpecHandler func(spec *specs.Spec, set container.GPUSet) error
+
+func init() {
+	vendors = make(map[string]SpecHandler)
```

Comment:
Don't think you need to do this in a `func init()`

Reply:
Need to double-check the order of initialization. I did the same as:
https://github.com/moby/moby/blob/master/daemon/graphdriver/driver.go#L148-L150